src/codegen/function.rs

```diff
@@ -23,9 +25,92 @@ use std::{
     result::Result as StdResult,
 };
 
+pub fn get_must_use_if_needed(
+    parent_type_id: Option<TypeId>,
+    analysis: &analysis::functions::Info,
+    in_trait: bool,
+    only_declaration: bool,
+    comment_prefix: &str,
+) -> Option<String> {
+    if analysis.ret.bool_return_is_error.is_some()
+        || analysis.ret.nullable_return_is_error.is_some()
+    {
+        return None;
+    }
+
+    let return_type = if !analysis.outs.is_empty() {
```

Comment:
Don't we have exactly this code somewhere else already?

Reply:
Kinda but not the same. :-/

Reply:
That's surprising :) We're just building the return type as a string here or not?

Reply:
Here we just check if there is one returned type and if it's the same as `self`. However I could check if we could somehow merge both functions.

Reply:
And can we? :)

Reply:
No. :'(